core/server/data/xml/sitemap/base-generator.js

```diff
@@ -55,13 +55,18 @@ _.extend(BaseSiteMapGenerator.prototype, {
         // Create all the url elements in JSON
         var self = this,
             nodes;
-        nodes = _.map(data, function (datum) {
+
+        nodes = _.reduce(data, function (nodeArr, datum) {
```

Comment:
I prefer to keep it the way it is, so that the validator ensures the content is correct even if the filter breaks or is not correct anymore. E.g. there's a fallback.

Long term, I'd like to replace this entire thing 😉